/*
 * Copyright (c) 2019 - now, Eggroll Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 *
 */

syntax = "proto3";

package com.webank.eggroll.core.egg;

message CountRequest {
    int64 value = 1;
}

message GetRequest {
    bytes key = 1;
}
message GetResponse {
    bytes key = 1;
    bytes value = 2;
    bool exists = 3;
}
message PutRequest {
    bytes key = 1;
    bytes value = 2;
}
message PutResponse {
    bytes key = 1;
    bytes value = 2;
    bool success = 3;
}

message GetAllRequest {
    int64 limit = 1;
}

message DeleteRequest {
    bytes key = 1;
}

message DeleteResponse {
    bytes key = 1;
    bool success = 2;
}

message MapPartitionsWithIndexRequest {
    bool shuffle = 1;
}

message ReduceResponse {
    int64 id = 1;
    bytes value = 2;
}

message AggregateRequest {
    bytes zero_value = 1;
}
message AggregateResponse {
    int64 id = 1;
    bytes value = 2;
}

message WithStoresResponse {
    int64 id = 1;
    bytes value = 2;
}